pkg/olm/installplan.go

```diff
@@ -0,0 +1,163 @@
+package olm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Builder provides a struct for installplan object from the cluster and a installplan definition.
+type Builder struct {
+	// installplan definition, used to create the installplan object.
+	Definition *v1alpha1.InstallPlan
+	// Created installplan object.
+	Object *v1alpha1.InstallPlan
+	// Used in functions that define or mutate installplan definition. errorMsg is processed
+	// before the installplan object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewBuilder creates new instance of Builder.
+func NewBuilder(apiClient *clients.Settings, name, nsname string) *Builder {
+	glog.V(100).Infof("Initializing new %s installplan structure", name)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &v1alpha1.InstallPlan{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the installplan is empty")
+
+		builder.errorMsg = "installplan 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the installplan is empty")
+
+		builder.errorMsg = "installplan 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// GetInstallPlanBuilder returns installplan (first from the list).
+func GetInstallPlanBuilder(apiClient *clients.Settings, nsname string) (*Builder, error) {
```

Comment:
Please re-name this function to ListInstallPlan and move it to the new file installplanlist.go.  This function has to return list > []*InstallPlanBuilder . You can use the first element of the returned list in your local test case code.